src/binder_data.c

```diff
@@ -570,6 +592,93 @@ binder_data_call_new_1_5(
     return call;
 }
 
+static
+BinderDataCall*
+binder_data_call_new_aidl(
+    GBinderReader* reader)
+{
+    BinderDataCall* call = binder_data_call_new();
+
+    gsize data_read;
+    gsize parcel_size = binder_read_parcelable_size(reader);
+    gsize initial_size = gbinder_reader_bytes_read(reader);
+    gint64 retry_time;
+
+    gbinder_reader_read_int32(reader, &call->status);
+    gbinder_reader_read_int64(reader, &retry_time);
+    // Is there better way to do this?
+    if (retry_time == G_MAXINT64) {
+        call->retry_time = G_MAXINT32;
+    } else if (retry_time < 0) {
+        call->retry_time = -1;
+    } else {
+        call->retry_time = (retry_time & 0xffffffff);
+    }
+    gbinder_reader_read_int32(reader, &call->cid);
+    gbinder_reader_read_uint32(reader, &call->active);
+    gbinder_reader_read_uint32(reader, &call->prot);
+    call->ifname = gbinder_reader_read_string16(reader);
+
+    // addresses
+    {
+        gint32 addresses_count;
+        guint i;
+        char** out;
+        char** ptr;
+        gbinder_reader_read_int32(reader, &addresses_count);
+
+        if (addresses_count < 0) {
+            addresses_count = 0;
+        }
+
+        out = g_new(char*, addresses_count + 1);
```

Comment:
Is it convention to not to care about small allocation failures so this should die in this case? ConnMan does it like it and it is fine, just asking how it is intended. Also, `g_new0()` to  get this initialized?

Reply:
Glib docs:
"If any call to allocate memory using functions g_new(), g_new0(), g_renew(), g_malloc(), g_malloc0(), g_malloc0_n(), g_realloc() and g_realloc_n() fails, the application is terminated. This also means that there is no need to check if the call succeeded."

Checking success of memory allocation is usually pointless with memory overcommit.

Reply:
Not really my main concern here. And it is about the coding conventions how to react.

Just not initializing stuff is kind of asking for trouble. Or messy code, as it has shown later too.

Reply:
Changed.